### PR TITLE
add etcd test

### PR DESCRIPTION
**Pull request by mhost39**
### Description

test etcd

### Related Issues

https://github.com/threefoldtech/js-sdk/issues/2255
### Checklist

- [ ] [Pre-commit hook is installed](https://github.com/threefoldtech/js-ng#pre-commit) to do formatting checks before committing code...etc
- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings

### CODE REVIEW

**Comment by mhost39** [comment]
image

File: tests/sals/automated_chatflows/test_solutions.py [hunk 1]
```diff
@@ -324,3 +324,29 @@ def test07_4to6gw(self):
...
+        - Check put a Hello by etcdctl.
+        """
+        self.info("Deploy etcd")
+        name = self.random_name()
+        etcd = deployer.deploy_etcd(solution_name=name, pool=self.pool_id, network=self.network_name,)
+        self.solution_uuid = etcd.solution_id
+
+        self.info("Check that etcd is reachable")
+        self.assertTrue(
+            self.wait(etcd.ip_addresses[0], port=2379, timeout=self.timeout),
```
Comment (waleedhammam) [inline tests/sals/automated_chatflows/test_solutions.py:344]: I think the ip is a private wg ip. how do we reach the etcd with ip ? shouldn't it be with domain ?

File: docs/tests/sals/automated_chatflows/test_solutions.md [hunk 2]
```diff
@@ -66,4 +66,14 @@ Test case for deploying a 4to6 GW.
...
+
+### test08_etcd
+
+Test case for deploying etcd.
+
+**Test Scenario**
+
+- Deploy etcd.
+- Check that etcd is reachable.
+- Check put a Hello by etcdctl.
```
Comment (waleedhammam) [inline docs/tests/sals/automated_chatflows/test_solutions.md:79]: we can add the get scenario too

Comment (mhost39) [reply docs/tests/sals/automated_chatflows/test_solutions.md:79]: this is for happy path 
anther scenario will be added with unhappy path

File: tests/sals/automated_chatflows/test_solutions.py (hunk 1 above)
Comment (waleedhammam) [reply tests/sals/automated_chatflows/test_solutions.py]: it's in the setup method

File: tests/sals/automated_chatflows/test_solutions.py [hunk 3]
```diff
@@ -324,3 +324,22 @@ def test07_4to6gw(self):
...
+        - Check put a Hello by etcdctl.
+        """
+        self.info("Deploy etcd")
+        name = self.random_name()
+        etcd = deployer.deploy_etcd(solution_name=name, pool=self.pool_id, network=self.network_name,)
+        self.solution_uuid = etcd.solution_id
+
+        self.info("Check put a Hello by etcdctl")
+        _, res, _ = j.sals.process.execute(
+            f"etcdctl --endpoints=http://{etcd.ip_addresses[0]}:2379 put from:{etcd.ip_addresses[0]} Hello"
```
Comment (AhmedHanafy725) [inline tests/sals/automated_chatflows/test_solutions.py:343]: I think the CI won't have `etcdctl` bin, no?

**Review by waleedhammam — APPROVED** [review]

**Review by xmonader — CHANGES_REQUESTED** [review]
I don't see etcd installed anywhere.

**Review by AhmedHanafy725 — APPROVED** [review]

**Review by xmonader — APPROVED** [review]

**Pull request merged**